annotations/views/rest_views.py

```diff
@@ -476,25 +480,87 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data,
                         status=status.HTTP_201_CREATED,
                         headers=headers)
-
     @list_route()
     def search(self, request, **kwargs):
         q = request.GET.get('search', None)
-        if not q:
-            return Response({'results': []})
-        pos = request.GET.get('pos', None)
-        concepts = goat.Concept.search(q=q, pos=pos, limit=50)
-
-        def _relabel(datum):
-            _fields = {
-                'name': 'label',
-                'id': 'alt_id',
-                'identifier': 'uri'
-            }
-
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
-
+        if cache.get(q.lower()) != None:
+            results = cache.get(q.lower())
+            print('Yes')
+        else:
+            if not q:
```

Comment:
if q is None, line 486 should already fail. this check needs to be done earlier.